ext/oss/sink.go

```diff
@@ -173,83 +124,56 @@ func (o *OSSSink) process() error {
 		}
 
 		err = o.DryRunable(func() error {
-			_, err = wh.Write(append(raw, '\n'))
-			if err != nil {
-				o.Logger().Error(fmt.Sprintf("failed to write to file"))
+			if err := o.handlers.Write(destinationURI, append(raw, '\n')); err != nil {
+				o.Logger().Error("failed to write to file")
 				return errors.WithStack(err)
 			}
-
 			recordCounter++
-			o.fileRecordCounters[destinationURI]++
-			if recordCounter%logCheckPoint == 0 {
-				o.Logger().Info(fmt.Sprintf("written %d records to file writer: %s", o.fileRecordCounters[destinationURI], destinationURI))
-			}
 			return nil
 		})
 		if err != nil {
 			return errors.WithStack(err)
 		}
+
 	}
 
 	if recordCounter == 0 {
 		o.Logger().Info(fmt.Sprintf("no records to write"))
 		return nil
 	}
 
-	// flush remaining records concurrently
-	funcs := []func() error{}
-	for destinationURI, wh := range o.writeHandlers {
-		funcs = append(funcs, func() error {
-			err := o.DryRunable(func() error {
-				if err := wh.Flush(); err != nil {
-					o.Logger().Error(fmt.Sprintf("failed to flush to %s", destinationURI))
-					return errors.WithStack(err)
-				}
-				o.Logger().Info(fmt.Sprintf("flushed %d records to file: %s", o.fileRecordCounters[destinationURI], destinationURI))
-				return nil
-			})
-			return errors.WithStack(err)
-		})
-	}
-	if err := o.ConcurrentTasks(funcs); err != nil {
+	// flush all write handlers
+	if err := o.DryRunable(o.handlers.Sync); err != nil {
```

Comment:
are we losing concurrent execution in sync? or can it be moved inside Sync function?